airbyte-integrations/connectors/source-shopify/unit_tests/graphql_bulk/test_job.py

```diff
@@ -341,3 +340,45 @@ def test_expand_stream_slices_job_size(
     list(stream.parse_response(test_bulk_response))
     # check the next slice
     assert stream.job_manager.job_size == adjusted_slice_size
+
+
+@pytest.mark.parametrize(
+    "job_response, error_type, expected",
+    [
+        (
+            "bulk_job_created_response",
+            ConnectionError,
+            "retrying Bad Request",
+        ),
+    ],
+    ids=[
+        "Failure after ConnectionError",
+    ],
+)
+def test_job_healthcheck_with_connection_errors(mocker, request, requests_mock, job_response, auth_config, error_type, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+
+    # Modify the sleep time for the test
+    stream.job_manager.concurrent_max_retry = 1
+    stream.job_manager.concurrent_interval_sec = 1
+    stream.job_manager.job_check_interval_sec = 1
+
+    # Get job_id from fixture.
+    job_id = request.getfixturevalue(job_response).get("data", {}).get("node", {}).get("id")
+
+    # Patch the method to get the right ID checks
+    if job_id:
+        mocker.patch("source_shopify.shopify_graphql.bulk.job.ShopifyBulkManager.job_get_id", value=job_id)
```

Comment:
I don't like this because we rely on patching internal methods. Why not requests_mock the "ShopifyBulk status" HTTP call?